integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonTableCompactor.scala

```diff
@@ -325,4 +339,65 @@ class CarbonTableCompactor(carbonLoadModel: CarbonLoadModel,
     }
   }
 
+  /**
+   * compact segments by global sort
+   */
+  def compactSegmentsByGlobalSort(
+      sparkSession: SparkSession,
+      carbonLoadModel: CarbonLoadModel,
+      carbonMergerMapping: CarbonMergerMapping): Array[(String, Boolean)] = {
+    val dataFrame = SegmentUtils.dataFrameOfSegments(
+      sparkSession,
+      carbonLoadModel.getCarbonDataLoadSchema.getCarbonTable,
+      carbonMergerMapping.validSegments)
+    val outputModel = getLoadModelForGlobalSort(
```

Comment:
please add comment why a new load model need to be created

Reply:
the old load model is used by local_sort.